ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/AdaptedV1WriteService.scala

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v2
+
+import java.time.Instant
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.daml.daml_lf_dev.DamlLf
+import com.daml.ledger.api.health.HealthStatus
+import com.daml.ledger.participant.state.v1
+import com.daml.lf.data.Time
+import com.daml.telemetry.TelemetryContext
+import com.google.rpc.code.Code
+import com.google.rpc.error_details.ErrorInfo
+import com.google.rpc.status.Status
+import io.grpc.{Metadata, StatusRuntimeException}
+
+import scala.jdk.CollectionConverters._
+
+class AdaptedV1WriteService(delegate: v1.WriteService) extends WriteService {
```

Comment:
Like for the `ReadService`, I recommend to add a disclaimer that this may not be a fully faithful implementation of the `v2.WriteService`.

Reply:
Added in #10254.